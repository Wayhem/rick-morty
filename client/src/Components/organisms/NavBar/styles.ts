import styled from 'styled-components'

export const Bar = styled.div`
  padding: 1rem;
  display: flex;
  flex-direction: row-reverse;
  background-color: ${props => props.theme.colors.white};
  box-shadow: 1px 1px 2px 2px ${props => props.theme.colors.transpBlack};
  align-items: center;
`

export const Username = styled.div`
  color: ${props => props.theme.colors.darkBlue};
  margin-right: 1rem;
`
