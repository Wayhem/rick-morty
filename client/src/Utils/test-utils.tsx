import React, { FC, ReactElement } from 'react'
import { render, RenderOptions, RenderResult, Queries } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { Provider } from 'react-redux'
import { MemoryRouter as Router } from 'react-router-dom'
import { store } from 'Store'
import { theme } from 'Config/styled'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const AllTheProviders: FC = ({ children }: { children?: any }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  </ThemeProvider>
)

AllTheProviders.defaultProps = {
  children: <></>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult<Queries, HTMLElement> => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'

export { customRender as render }
