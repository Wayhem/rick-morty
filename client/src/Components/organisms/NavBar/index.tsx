import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useLocation, useHistory } from 'react-router-dom'
import { useTheme } from 'styled-components'
import { Bar, Username, Spacer } from 'Components/organisms/NavBar/styles'
import OutlinedButton from 'Components/atoms/outlinedButton'
import routes from 'Config/routes'
import authTypes from 'Store/types/authTypes'
import { State } from 'Store/state'

const NavBar = (): JSX.Element => {
  const dispatch = useDispatch()
  const theme = useTheme()
  const location = useLocation()
  const history = useHistory()
  const username = useSelector<State, string>((state: State) => state.auth.username)

  const logout = () => dispatch({ type: authTypes.LOGOUT })

  const goToFavorites = () => history.push(routes.favorites)
  const goHome = () => history.push(routes.home)

  return (
    <Bar>
      <OutlinedButton onClick={logout}>Log out</OutlinedButton>
      <Spacer>
        {location.pathname !== routes.favorites && (
          <OutlinedButton
            color={theme.colors.lighterDarkBlue}
            onClick={goToFavorites}
          >
            Go to favorites
          </OutlinedButton>
        )}
        {location.pathname !== routes.home && (
          <OutlinedButton
            color={theme.colors.lighterDarkBlue}
            onClick={goHome}
          >
            Go to characters list
          </OutlinedButton>
        )}
      </Spacer>
      <Username>Hello, {username}</Username>
    </Bar>
  )
}

export default NavBar
