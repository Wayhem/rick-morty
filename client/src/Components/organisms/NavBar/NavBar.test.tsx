import React from 'react'
import NavBar from 'Components/organisms/NavBar'
import { render } from 'Utils/test-utils'

test('NavBar renders and matches snapshot', () => {
  const { container } = render(<NavBar />)
  expect(container).toMatchSnapshot()
})
