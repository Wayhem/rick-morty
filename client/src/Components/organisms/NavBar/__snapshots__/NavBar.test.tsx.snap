// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders and matches snapshot 1`] = `
<div>
  <div
    class="sc-bdfBwQ gFFhXP"
  >
    <button
      class="sc-hKgILt geIEcq"
    >
      Log out
    </button>
    <div
      class="sc-dlfnbm epLfUN"
    >
      <button
        class="sc-hKgILt dtgEiG"
        color="#647183"
      >
        Go to favorites
      </button>
    </div>
    <div
      class="sc-gsTCUz KDXhW"
    >
      Hello, 
      
    </div>
  </div>
</div>
`;
